sql/src/main/java/io/crate/expression/symbol/format/SymbolPrinter.java

```diff
@@ -182,9 +212,92 @@ private String anyOperatorName(String functionName) {
             return functionName.substring(4).replace('_', ' ').toUpperCase(Locale.ENGLISH);
         }
 
+        private void printMatchPredicate(Function matchPredicate, SymbolPrinterContext context) {
+            StringBuilder sb = context.builder;
+            // Due to the match predicate being a hack and not an actual function, we have to manually print it here
+            sb.append("MATCH(");
+            List<Symbol> arguments = matchPredicate.arguments();
+            // first argument (cols)
+            Map columnBootMap = (Map) ((Literal) arguments.get(0)).value();
+            sb.append("(");
+            Iterator<Map.Entry<String, Object>> entryIterator = columnBootMap.entrySet().iterator();
+            while (entryIterator.hasNext()) {
```

Comment:
would it make sense to split this method up a bit?

Reply:
It's rather lengthy due to the complexity that comes with printing the MATCH clause. There are some comments to mark the different parts. On the one hand, breaking it up would make it more readable but on the other hand we would end up with parts which are not used independently of each other.

Reply:
Maybe introduce a dedicated `MatchPrinter` ?

Reply:
Could do that but there is already a dedicated method `printMatchPredicate`. I don't see the added benefit of having an extra class.

Reply:
Would help in being able to introduce some private methods to split up the code here without cluttering the SymbolPrinter